segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/delegationtoken/TokenVerifierImpl.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host.delegationtoken;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.pravega.client.auth.PravegaAuthHandler;
+import io.pravega.client.auth.PravegaAuthenticationException;
+import io.pravega.segmentstore.server.host.stat.AutoScalerConfig;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class TokenVerifierImpl implements DelegationTokenVerifier {
+    private final AutoScalerConfig config;
+
+    public TokenVerifierImpl(AutoScalerConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public boolean verifyToken(String resource, String token, PravegaAuthHandler.PravegaAccessControlEnum expectedLevel) {
+        if (config.isAuthEnabled()) {
+            try {
+                Jws<Claims> claims = Jwts.parser()
+                                         .setSigningKey(config.getTokenSigningKey().getBytes())
+                                         .parseClaimsJws(token);
+                Optional<Map.Entry<String, Object>> matchingClaim = claims.getBody().entrySet().stream().filter(entry -> resource.startsWith(entry.getKey())
+                        && expectedLevel.compareTo(PravegaAuthHandler.PravegaAccessControlEnum.valueOf(entry.getValue().toString()))
+                        < 0).findFirst();
+                if (matchingClaim.isPresent()) {
+                    log.debug("Found a matching claim {} for resource {}", matchingClaim, resource);
+                    return false;
```

Comment:
Isn't the logic inverted?   (meaning matching claim should return true?)

Reply:
Fixed. Thanks for catching this.